homeassistant/components/rfxtrx/config_flow.py

```diff
@@ -342,6 +344,17 @@ async def _async_replace_device(self, replace_device):
 
         device_registry.async_remove_device(old_device)
 
+    def _can_add_device(self, new_rfx_obj):
+        """Check if device does not already exist."""
+        new_device_id = get_device_id(new_rfx_obj.device)
+        for packet_id, entity_info in self._config_entry.data[CONF_DEVICES].items():
+            rfx_obj = get_rfx_object(packet_id)
+            device_id = get_device_id(rfx_obj.device, entity_info.get(CONF_DATA_BITS))
+            if new_device_id == device_id:
+                return False
+
+        return True
```

Comment:
Just thinking out loud: is all this really necessary?

In the Plugwise config_flow we use this:
```
            for entry in self._async_current_entries():
                if entry.data.get(CONF_HOST) == user_input[CONF_HOST]:
                    return self.async_abort(reason="already_configured")
```
Can't you check whether "device_id" is already present in the config_entries? The config_entries are stored in the file config/.storage/core.config_entries, I'd suggest you have a look in that file to understand what I mean.

Reply:
What do you want me to find in core.config_entries? The entries there are nothing more than a storage for data we write in config flow.

Note that we are not in config flow here but options flow and that we are not preventing duplicate config entries.

Reply:
Ok, understood.
Confusing, you don't see the code around the changed parts, I thought this was in a step-function.

Reply:
In that case, please ignore my comment.

Reply:
No problem :)